plugin/health_check.go

```diff
@@ -137,10 +147,10 @@ func (job *PluginHealthCheckJob) Cancel() {
 // shouldDeactivatePlugin determines if a plugin needs to be deactivated after certain criteria is met.
 //
 // The criteria is based on if the plugin has consistently failed during the configured number of restarts, within the configured time window.
-func shouldDeactivatePlugin(rp registeredPlugin) bool {
-	if len(rp.failTimeStamps) >= HEALTH_CHECK_RESTART_LIMIT {
-		index := len(rp.failTimeStamps) - HEALTH_CHECK_RESTART_LIMIT
-		t := rp.failTimeStamps[index]
+func shouldDeactivatePlugin(failedTimestamps []time.Time) bool {
+	if len(failedTimestamps) >= HEALTH_CHECK_RESTART_LIMIT {
+		index := len(failedTimestamps) - HEALTH_CHECK_RESTART_LIMIT
+		t := failedTimestamps[index]
 		now := time.Now()
 		elapsed := now.Sub(t)
 		if elapsed <= HEALTH_CHECK_DISABLE_DURATION {
```

Comment:
just wondering, why do we have to do this check isn't this always true since the time between now and last failure will always be less than HEALTH_CHECK_DISABLE_DURATION because we ping every 30 seconds?

Can we rely on number of failures rather than the timestamp? would make the logic alot simpler :)

Reply:
> just wondering, why do we have to do this check isn't this always true since the time between now and last failure will always be less than HEALTH_CHECK_DISABLE_DURATION because we ping every 30 seconds?

The reasoning behind the timestamps is that a failure might be happening every 10 minutes or so, maybe because of an endpoint that occasionally gets called is panicking. The consensus was that if the plugin fails three times within one hour, then we disable the plugin. This was the most straightforward implementation I could think of that satisfies the requirement.

Reply:
ahhh I see. The implementation make sense now :) Thanks for the clarification.

Maybe worth adding a comment here something to the effect of 
`// Disable plugin if last HEALTH_CHECK_RESTART_LIMIT failures happen within the HEALTH_CHECK_DISABLE_DURATION time limit`

also i'd suggest using `time.Since` which is just a shorthand for `time.Now().Sub(t)`, but makes it a tiny bit easier to read.

https://golang.org/pkg/time/#Since